samples/caf/README.rst

```diff
@@ -0,0 +1,119 @@
+
+The sample adds its own module, named `led_state`, that is:
+
+* Listening to button events send by the CAF Buttons module.
+* Sending led effect events to the CAF LED module.
+
+.. note:
+    For more about adding your own module refer to `Adding custom modules` section in :ref:`caf_overview`.
+
+.. _caf_sample_requirements:
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf52840dk_nrf52840, nrf5340dk_nrf5340_cpuapp
+
+User interface
+**************
+
+Button 1:
+   If pressed, the LED effect displayed on the LED changes to the next defined LED effect.
+
+Button 2:
+   If pressed, the application swaps to the next available LED.
+
+LEDs:
+   Based on the buttons presses displayed LED effects would change for chosen LED.
+
+Configuration
+*************
+
+The sample supports the configuration for boards defined in :ref:`caf_sample_requirements` section of this documentation.
+
+The configuration for each board is different, because on board LEDs are connected to different GPIO pins.
+For more about configuration of each module, refer to its documentation:
+
+* :ref:`caf_leds` for LEDs module.
+* :ref:`caf_buttons` for buttons module.
+
+Building and running
+********************
+.. |sample path| replace:: :file:`samples/caf`
+
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+
+After you program the development kit you will see that the first available LED is set to on.
+
+By pressing the buttons user can change the displayed effect for the LED or swap to another LED.
+The sample defines a few LED effects that could be displayed on LEDs.
+The LED effects are defined in `led_state_def.h` file that is a part of the configuration.
+The bellow code snippet show the default led effetcts configuration of the sample.
+
+  .. code-block:: none
+
+	static const struct led_effect led_effect[] = {
+		[LED_EFFECT_ID_OFF] = LED_EFFECT_LED_OFF(),
+		[LED_EFFECT_ID_BLINK] = LED_EFFECT_LED_BLINK(200, LED_COLOR(255, 255, 255)),
+		[LED_EFFECT_ID_BREATH] = LED_EFFECT_LED_BREATH(200, LED_COLOR(255, 255, 255)),
+		[LED_EFFECT_ID_CLOCK] = LED_EFFECT_LED_CLOCK(3, LED_COLOR(255, 255, 255)),
+	};
+
+When testing the sample you can change the defined effects or add your own effects to the `led_effect` table.
+
+After programming the sample to your development kit, you can test it by performing the following steps:
+
+1. Press Button 1 of the development kit to swap to the next LED effect of the LED1.
+   The LED color should start to change each 200ms between chosen color and the LED turned off.
+#. Press Button 2 of the development kit to swap to the next LED.
+   The LED2 of the development kit should be set to on, indicating its currently being controlled by the application.
+#. Press Button 1 and see the LED2 starts blinking similarly as LED1.
+#. Press Button 1 again and see the LED2 starts breathing with 200ms period.
```

Comment:
~400 ms period :) As mentioned before ~200 ms is time for either led brightness increase or decrease

We should also fix doxygen doc for LED breathing effect too (_period no longer refers to a single substep)